src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1062,31 +1107,11 @@ private enum AssignmentKind
 
             if (value == null ||
                 value.WasCompilerGenerated ||
-                !targetType.HasType ||
-                targetType.Type.IsValueType)
+                !ShouldReportNullableAssignment(targetType, valueType.State))
```

Comment:
Are there any behavioral changes from this change?

Reply:
No, this is just extracting the logic so that it can be called from the new checks.